### PR TITLE
DrivAerML model capacity sweep: 2L/128d vs 3L/192d vs 4L/256d + large anchors

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

On AirfRANS, the default 3L/192d model with AdamW was optimal. On DrivAerML, the 3D geometry might benefit from more capacity since the input space is higher-dimensional (3D coordinates) and the training set is larger (394 cars). We sweep 3 model sizes to find the capacity sweet spot for DrivAerML.

## Current Baseline

- DrivAerML: **no baseline yet** — target to beat: surface_rel_l2_pct < 3.71% (Transolver-3)

## Instructions

Run 3 parallel trials with `CUDA_VISIBLE_DEVICES`. Use `--wandb_group radford-drivaerml-round1` for all runs. Use AdamW lr=5e-4, batch_size=1, cosine_t_max=30.

### Trial 0 — GPU 0: Default 3L/192d/3H (baseline capacity)
```bash
CUDA_VISIBLE_DEVICES=0 python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine_t_max 30 --batch-size 1 --model-layers 3 --model-hidden-dim 192 --model-heads 3 --model-slices 96 --wandb_group radford-drivaerml-round1 --wandb_name mitsuha/drivaerml-cap-3l192d --agent mitsuha
```

### Trial 1 — GPU 1: Larger 4L/256d/4H
```bash
CUDA_VISIBLE_DEVICES=1 python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine_t_max 30 --batch-size 1 --model-layers 4 --model-hidden-dim 256 --model-heads 4 --model-slices 96 --wandb_group radford-drivaerml-round1 --wandb_name mitsuha/drivaerml-cap-4l256d --agent mitsuha
```

### Trial 2 — GPU 2: Smaller 2L/128d/2H (faster, more epochs)
```bash
CUDA_VISIBLE_DEVICES=2 python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine_t_max 30 --batch-size 1 --model-layers 2 --model-hidden-dim 128 --model-heads 2 --model-slices 96 --wandb_group radford-drivaerml-round1 --wandb_name mitsuha/drivaerml-cap-2l128d --agent mitsuha
```

### Trial 3 — GPU 3: Larger anchor budget (geometry_supernodes=8192, surface_anchor_points=16000)
```bash
CUDA_VISIBLE_DEVICES=3 python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine_t_max 30 --batch-size 1 --geometry-supernodes 8192 --surface-anchor-points 16000 --wandb_group radford-drivaerml-round1 --wandb_name mitsuha/drivaerml-anchor-large --agent mitsuha
```

Run all 4 in parallel (background each with `&`). Wait for all to finish.

**Notes:**
- If Trial 1 (4L/256d) or Trial 3 (large anchors) OOM, reduce to batch_size=1 with fewer supernodes
- The primary metric is `val_primary/surface_rel_l2_pct` (lower is better)
- Also report wall time per epoch for each trial — throughput varies by model size

## Results

_To be filled by student after runs complete._

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT mitsuha:

## OOM on all 4 trials — adding surface point cap

All 4 trials OOM'd on 95GB GPUs (even the smallest 2L/128d model used 85GB before crashing). Root cause: `drivaerml_train_surface_points` and `drivaerml_eval_surface_points` default to `0`, which loads the **full unsubsampled mesh** per car. DrivAerML surface meshes are very large (hundreds of thousands of points).

**Also note:** `--geometry-supernodes` and `--surface-anchor-points` only affect the `reference_abupt` model, not `senpai_transolver`. Trial 3's settings had no effect on memory.

**Fix:** Adding `--drivaerml-train-surface-points 32000 --drivaerml-eval-surface-points 32000` to all trials. This aligns with NeuralCFD's 40k subsample and AB-UPT's 16k anchors. At 32k points, estimated peak memory is ~2-3GB for activations, well within 95GB.

Relaunching all 4 trials now. Timestamp: 2026-04-20 ~19:30 UTC.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2442 because Superseded: mitsuha reassigned to #2607 (DrivAerML 4L/256d T_max=20).
